.github/workflows/ci.yml

```diff
@@ -0,0 +1,40 @@
+name: CI
+
+on: [push, pull_request]
+
+jobs:
+  test:
+    name: Node.js v${{ matrix.nodejs }}
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        nodejs: [12, 14]
+    env:
+      CI: true
+    steps:
+    - uses: actions/checkout@v2
+    - uses: actions/setup-node@v1
+      with:
+        node-version: ${{ matrix.nodejs }}
+
+    - name: Install
+      run: |
+        npm ci
+    - name: Compiles
+      run: npm run build
+    
+    - name: Check un-tracked changes in /dist using git porcelain command
+      run: |
+        if [[ `git status ./dist --porcelain` ]]
```

Comment:
I'm not sure this really works. After adding a new, untracked file, `git status --porcelain` still exits with `0`. 🤔

Reply:
Some people propose checking `[[ -z "$(git status --porcelain)" ]]`.

Reply:
> Some people propose checking [[ -z "$(git status --porcelain)" ]].

what is the difference (apart from checking the root dir and not just ./dist)?
we probably want to scope to ./dist to make sure everything is built/compiled and don't care about the rest 🤔

Reply:
> I'm not sure this really works. After adding a new, untracked file, git status --porcelain still exits with 0. 🤔

bust at least ./dist/index.js or some sub-files should change (due to changed imports). So I'd think this is good enough. Tested it in a supped-to-be-breaking commit with a changed file.

Reply:
> what is the difference (apart from checking the root dir and not just ./dist)?

The difference is it doesn't check the exit status, which still seems to be `0` when new untracked files are added (e.g. because you added a new TypeScript source file, but didn't update `dist/`).

Reply:
Ah no, scratch that. You have backticks there. But you never check the output is empty, do you?

Reply:
<img width="725" alt="Screenshot 2021-02-18 at 22 45 02" src="https://user-images.githubusercontent.com/206108/108425614-fdffc700-723a-11eb-8b80-dfbbfccda46a.png">

To end this argument (in a good way, I hope) I just tried it. Just pushed a commit (will be reverted soon) which introduces a test file that was not compiled. all builds are red with the above error

Reply:
I didn't really perceive this as an argument. I apologize if I came across rough or rude. I was only dropping in while working on client work with a need to be responsive there and didn't have a whole lot of time to consider my wording/responses here or to try out things myself. 💚

Reply:
ah, sorry. not taken as rude :) I meant discussion, but my german-thinking brain thought argument is a good word since we exchange arguments 🤔  💚

Reply:
👆 please don't be sorr,y no reason for that. I'm really thankful for your review(s)